pandas/io/parsers/base_parser.py

```diff
@@ -202,6 +208,25 @@ def __init__(self, kwds):
 
         self.handles: Optional[IOHandles] = None
 
+        # Bad line handling
+        on_bad_lines = kwds.get("on_bad_lines")
+        if on_bad_lines is not None:
+            if on_bad_lines == "error":
+                self.on_bad_lines = self.BadLineHandleMethod.ERROR
+            elif on_bad_lines == "warn":
+                self.on_bad_lines = self.BadLineHandleMethod.WARN
+            elif on_bad_lines == "skip":
+                self.on_bad_lines = self.BadLineHandleMethod.SKIP
+            else:
+                raise ValueError(f"Argument {on_bad_lines} is invalid for on_bad_lines")
+        else:
+            if kwds.get("error_bad_lines"):
```

Comment:
we need to raise if on_bad_lines is not None and (error_bad_lines or warn_bad_lines)

Reply:
I have a really sketchy solution involving setting on_bad_lines to None at first. This is so we can tell if it is set manually or not and raise this. Then we alias to error, so default is technically still error even if function signature doesn't say so.